common/extensions/api/web_discovery.json

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+[
+  {
+    "namespace": "webDiscovery",
+    "description": "Includes Web Discovery functions for the Brave extension",
+    "compiler_options": {
+      "implemented_in": "brave/browser/extensions/api/web_discovery_api.h"
+    },
+    "functions": [
+      {
+        "name": "retrieveBackupResults",
+        "type": "function",
+        "description": "Requests search results from backup provider and returns SERP HTML",
+        "parameters": [
+          {
+            "name": "url",
+            "type": "string"
+          }
+        ],
+        "returns_async": {
```

Comment:
Would it be possible to return more information here? In particular:
- status code
- response headers
- response url
- if the request was redirected or not (might not be as critical but good to have)

Reply:
Would it be possible to implement some kind of "AbortController" via this API? (if yes, I think it's not super critical but we do rely on this ability of the `fetch` API currently in WDP)

Reply:
Additionally, having a way to enforce a timeout would be nice to have.

Reply:
Added status code. Five second timeout is enforced.